com.unity.render-pipelines.core/Editor/Lighting/IESEngine.cs

```diff
@@ -0,0 +1,405 @@
+using System.IO;
+using Unity.Collections;
+using UnityEditor;
+using UnityEditor.Experimental.AssetImporters;
+using UnityEngine;
+
+namespace UnityEditor.Rendering
+{
+    // Photometric type coordinate system references:
+    // https://www.ies.org/product/approved-method-guide-to-goniometer-measurements-and-types-and-photometric-coordinate-systems/
+    // https://support.agi32.com/support/solutions/articles/22000209748-type-a-type-b-and-type-c-photometry
+
+    [System.Serializable]
+    public class IESEngine
+    {
+        const float k_HalfPi = 0.5f * Mathf.PI;
+        const float k_TwoPi  = 2.0f * Mathf.PI;
+
+        IESReader m_IesReader = new IESReader();
+
+        public string FileFormatVersion { get => m_IesReader.FileFormatVersion; }
+
+        TextureImporterType m_TextureGenerationType = TextureImporterType.Cookie;
+        public TextureImporterType TextureGenerationType
+        {
+            set { m_TextureGenerationType = value; }
+        }
+
+        public string ReadFile(string iesFilePath)
+        {
+            if (!File.Exists(iesFilePath))
+            {
+               return "IES file does not exist.";
+            }
+
+            string errorMessage;
+
+            try
+            {
+                errorMessage = m_IesReader.ReadFile(iesFilePath);
+            }
+            catch (IOException ioEx)
+            {
+                return ioEx.Message;
+            }
+
+            return errorMessage;
+        }
+
+        public string GetKeywordValue(string keyword)
+        {
+            return m_IesReader.GetKeywordValue(keyword);
+        }
+
+        public string GetPhotometricType()
+        {
+            switch (m_IesReader.PhotometricType)
+            {
+                case 3: // type A
+                    return "Type A";
+                case 2: // type B
+                    return "Type B";
+                default: // type C
+                    return "Type C";
+            }
+        }
+
+        public (float, string) GetMaximumIntensity()
+        {
+            if (m_IesReader.TotalLumens == -1f) // absolute photometry
+            {
+                return (m_IesReader.MaxCandelas, "Candelas");
+            }
+            else
+            {
+                return (m_IesReader.TotalLumens, "Lumens");
+            }
+        }
+
+        public (string, Texture) GenerateCubeCookie(TextureImporterCompression compression, int textureSize)
+        {
+            int width  = 2*textureSize;
```

Comment:
note: the convention is "2 * textureSize"  not "2*textureSize", please check in all your PR :)

Reply:
Also all public functions need to be documented

Reply:
360f/(width - 1); => 360f / (width - 1);